libraries/core/src/descriptor/mod.rs

```diff
@@ -94,19 +99,26 @@ impl Descriptor {
         Ok(flowchart)
     }
 
-    pub async fn read(file: &Path) -> eyre::Result<Descriptor> {
-        let descriptor_file = tokio::fs::read(file)
+    pub async fn read(path: &Path) -> eyre::Result<Descriptor> {
+        let buf = tokio::fs::read(path)
             .await
             .context("failed to open given file")?;
-        let descriptor: Descriptor =
-            serde_yaml::from_slice(&descriptor_file).context("failed to parse given descriptor")?;
-        Ok(descriptor)
+        Descriptor::parse(buf, path)
     }
 
-    pub fn blocking_read(file: &Path) -> eyre::Result<Descriptor> {
-        let descriptor_file = std::fs::read(file).context("failed to open given file")?;
-        let descriptor: Descriptor =
-            serde_yaml::from_slice(&descriptor_file).context("failed to parse given descriptor")?;
+    pub fn blocking_read(path: &Path) -> eyre::Result<Descriptor> {
+        let buf = std::fs::read(path).context("failed to open given file")?;
+        Descriptor::parse(buf, path)
+    }
+
+    pub fn parse(buf: Vec<u8>, path: &Path) -> eyre::Result<Descriptor> {
+        let mut descriptor: Descriptor =
+            serde_yaml::from_slice(&buf).context("failed to parse given descriptor")?;
+        let base = path.canonicalize().unwrap().parent().unwrap().to_owned();
+        descriptor.base_path = base;
```

Comment:
If the `base_path` field is filled with a user-provided path, we should not override it. Alternatively, we could make the `base_path` an argument to `check_dataflow` instead of a struct field if we want to use it only internally.

Reply:
Ok. I made it an argument of `check` and `check_dataflow`